src/serve.rs

```diff
@@ -96,10 +140,45 @@ impl ServeSystem {
 pub struct State {
     /// The path to the index.html file.
     pub index: Arc<PathBuf>,
+
+    pub ws: WebSockets,
+    pub inject_reload_script: bool,
+}
+
+const RELOAD_SCRIPT: &'static str = include_str!("autoreload.js");
+
+fn inject_reload_script(html: &Vec<u8>) -> Option<Vec<u8>> {
+    let mut script = std::io::Cursor::new(format!("<script>{}</script>", RELOAD_SCRIPT).as_bytes().to_vec());
+    let parsed_script = kuchiki::parse_html().from_utf8().read_from(&mut script).unwrap();
```

Comment:
After cutting over to nipper, this unwrap may no longer be applicable, but if it is, let's remove it in favor of just logging an error in the console & then returning `None`.